### PR TITLE
Add custom CSS for shiny bikes

### PR DESCRIPTION
**Pull request by seirl**
Some bikes are shiny. It's good to be notified when a nearby station has a shiny bike, so you can be sure to grab it.

image

image

### CODE REVIEW

**Review by delroth — APPROVED** [review]
The list matches my own list of shiny publibikes too, haven't seen any other than those 3 around Zürich. I'll let @zopieux review the HTML/CSS changes if he cares - it LGTM but I don't have very strong feelings about the frontend code at this point.

File: app.ts [hunk 1]
```diff
@@ -255,6 +262,10 @@ function FormatEBike(eb: EBike): string {
                         const $span = document.createElement('span');
```
Comment (zopieux) [inline app.ts:256]: Remove the `map` above and `FormatEBike()`, then:
```js
forEach(ebike => {
const $span = document.createElement('span');

const $name = document.createElement('span');
$span.appendChild($name);
$name.textContent = ebike.name;

if (SHINY_BIKES.includes(ebike.name)) { $name.classList.add("text-shiny"); }

if (ebike.battery != -1) {
  const $bat =  document.createElement('span');
  $bat.textContent = ` (${ebike.battery}%)`;
  $span.appendChild($bat);
}

$best_ebikes.appendChild($span);
})
```

**Review by zopieux — CHANGES_REQUESTED** [review]
Neat, thanks!

File: app.ts (hunk 1 above)
Comment (seirl) [reply app.ts:256]: Done (small changes to make it actually work).

File: index.html [hunk 2]
```diff
@@ -138,6 +138,32 @@
...
+
+        .text-shiny {
+            position: relative;
+        }
+
+        .text-shiny::before {
+            content: "";
+            position: absolute;
+            top: 0;
+            left: 0;
```
Comment (zopieux) [inline index.html]: Can you try to either `margin: 0 2px;` or `padding` this please? There is no breathing room for the animation to shine.

**Review by zopieux — APPROVED** [review]

File: index.html (hunk 2 above)
Comment (zopieux) [reply index.html]: Achktually, simpler, and minor changes to gradient to make it fancier (in my humble opinion):

[Preview](https://codepen.io/zopieux/pen/YzmMRJd)

```css
/* remove position-relative shit above */

        .text-shiny {
            padding: 3px 4px;
            background-image: linear-gradient(
              45deg,
              goldenrod, palegoldenrod, goldenrod
            );
            background-size: 200% 100%;
            animation: shinyEffect 4s ease-in-out infinite;
            border-radius: 3px;
        }
```

Comment (zopieux) [reply index.html]: Ok hear me out (the webkit stuff is understood by firefox as well):

```css
        .text-shiny {
           padding: 3px 4px;
           font-weight: bold;
           background-color:         goldenrod;
              background-image: linear-gradient(
              to right,
              transparent, rgba(238,232,170, 0.8), transparent
            );
            background-size: 200% 100%;
            animation: shinyEffect 3s linear infinite;
            border-radius: 3px;
            -webkit-background-clip: text;
            -webkit-text-fill-color: transparent;
        }

@keyframes shinyEffect {
            0% { background-position: -100% center; }
            50% { background-position: 100% center; }
            100% { background-position: -100% center; }
        }
```

**Pull request merged**